e2e/test_operator_upgrades_with_chaos/operator_upgrades_with_chaos_test.go

```diff
@@ -164,9 +164,6 @@ var _ = Describe("Operator Upgrades with chaos-mesh", Label("e2e", "pr"), func()
 				Consistently(func() bool {
 					return fdbCluster.GetCluster().Status.RunningVersion == beforeVersion
 				}).WithTimeout(2 * time.Minute).WithPolling(2 * time.Second).Should(BeTrue())
-			} else {
-				// Simulate the 2 minute wait otherwise the auto replacement might not succeed in time.
-				time.Sleep(2 * time.Minute)
```

Comment:
That sleep time was useless in the case of protocol compatible upgrades as the operator would move ahead and recreate the Pods for the new desired version. In addition below we only check for Pod removal in the case of version incompatible upgrades (because in the case of the protocol compatible upgrade the upgrade process is not blocked by a missing process and if that partitioned pod is recreated the chaos-mesh partition is lost.